### PR TITLE
Fee restructure

### PR DESCRIPTION
**Pull request by marcushanses**
- [x] https://github.com/intergiro/model-acquirer/pull/305

### CODE REVIEW

File: Settlement/newFee.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export type newFee = {
```
Comment (Magnus93) [inline Settlement/newFee.ts:1]: Should have uppercase `N` in `NewFee`.

Alternative:
Put the old `Fee` type in a folder called `V1` So old fee could be referred to as `Settlement.V1.Fee` and this fee could be `Settlement.Fee`

**Review by Magnus93 — CHANGES_REQUESTED** [review]

File: Settlement/newFee.ts (hunk 1 above)
Comment (marcushanses) [reply Settlement/newFee.ts]: https://github.com/intergiro/model-acquirer/pull/306/commits/237c31dfc4a32a9a32b8ed6f2abc969b653df869

**Review by robinnedergard — CHANGES_REQUESTED** [review]
EDIT: FIXED!

original message:
On a capture test I got this weird change in test results in worker-settlement when testing with this branch. Do you know what it might be caused by? A quick check makes me think that all/many scheme/interchange fees might be reversed in the test results?
```
- Expected  - 4
+ Received  + 4

"fee": Object {
    -         "scheme": 12,
    +         "interchange": -12,
    +         "intergiro": 61.99,
    +         "network": 0,
              "total": 49.99,
            },
```

File: Settlement/Fee.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	export function upgrade(value: number | { scheme: number; total: number } | Fee): Fee {
+		let result: Fee
+		if (is(value))
+			result = value
+		else if (typeof value == "number")
+			result = { interchange: 0, intergiro: 0, total: value }
+		else
+			result = {
+				interchange: value.total - value.scheme,
+				intergiro: value.scheme,
```
Comment (robinnedergard) [inline Settlement/Fee.ts:30]: Wrong, should be:
```
interchange: value.scheme,
intergiro: value.total - value.scheme,
```
Also use isoly.Currency.subtract() in this case as we don't have more than currency dependant decimals in this case.

**Review by robinnedergard — CHANGES_REQUESTED** [review]

File: Settlement/Fee.ts (hunk 2 above)
Comment (marcushanses) [reply Settlement/Fee.ts:30]: Ah, thank you for noticing!

**Review by robinnedergard — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**